### DISCUSSION TITLE
Any way to map cmd to ctrl?

### DISCUSSION THREAD

**Discussion by AlgusDark** [wezterm/wezterm#2206] [Q&A] [answered]
I want to map my cmd key to be ctrl. Something I did was a for loop for all the letters but, is there an easier way?

```lua
local alphabet = "!\"#$%&'()*+,-./0123456789:;<=>?@[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~"
local length = #alphabet
local keys = {}
for i = 1,length do
  local char = string.sub(alphabet, i, i)
  table.insert(keys, {key = char, mods= "SUPER", action={SendKey={key=char, mods="CTRL"}}})
end

for k,v in pairs(keys)
do
  table.insert(config.keys, v);
end
```

**Comment by wez** [comment, marked as answer]
I would suggest doing this in your OS settings.
See this for a (tiny bit) more info and a suggestion:
* https://github.com/wez/wezterm/issues/1777#issuecomment-1079944150

**Reply by AlgusDark** [reply]
The things is that I already use QMK for remapping my keyboard layout. The only thing I want to do is to remap the cmd to Ctrl on the terminal. I know I can do it with karabiner per app, or create another layout in QMK, but the former is not an option on my current laptop for permissions from my working laptop and the latter is not the ideal because of the extra steps.

I thought there could be an easy way of doing that inside of the terminal, mostly because is useful when using vim as I shouldn't thing on different key for using Ctrl.

I'll dig to see if there are other solutions. For windows and Linux is not a problem as the Windows/super key is not the main key for those OS.

**Reply by MuhammedZakir** [reply]
> I'll dig to see if there are other solutions. [...]

1. You can use Hammerspoon, BetterTouchTool, etc. to remap keys directly.
2. If you want to remap at a lower-level, you can use `hidutil`. Execute it whenever the frontmost application changes. For that, you can use any of the above apps, `yabai`, or any other tools that can watch windows and execute commands when some property changes.

**Reply by AlgusDark** [reply]
Thanks @MuhammedZakir Hammerspoon is a very cool tool and I'm digging into it so I can do this and is going to be an opportunity for using yabai :)
